flake.nix

```diff
@@ -8,183 +8,166 @@
-        src = filteredSource;
-        # disable fetching and building of tree-sitter grammars in the helix-term build.rs
+
+        # Helix attempts to reach out to the network and get the grammars. Nix doesn't allow this.
         HELIX_DISABLE_AUTO_GRAMMAR_BUILD = "1";
-        buildInputs = [stdenv.cc.cc.lib];
-        nativeBuildInputs = [pkgs.installShellFiles];
-        # disable tests
+
+        # So Helix knows what rev it is.
+        HELIX_NIX_BUILD_REV = self.rev or self.dirtyRev or null;
+
         doCheck = false;
-        meta.mainProgram = "hx";
-      };
-      cargoArtifacts = craneLibMSRV.buildDepsOnly commonArgs;
-    in {
-      packages = {
-        helix-unwrapped = craneLibStable.buildPackage (commonArgs
-          // {
-            cargoArtifacts = craneLibStable.buildDepsOnly commonArgs;
-            postInstall = ''
-              mkdir -p $out/share/applications $out/share/icons/hicolor/scalable/apps $out/share/icons/hicolor/256x256/apps
-              cp contrib/Helix.desktop $out/share/applications
-              cp logo.svg $out/share/icons/hicolor/scalable/apps/helix.svg
-              cp contrib/helix.png $out/share/icons/hicolor/256x256/apps
-              installShellCompletion contrib/completion/hx.{bash,fish,zsh}
-            '';
-            # set git revision for nix flake builds, see 'git_hash' in helix-loader/build.rs
-            HELIX_NIX_BUILD_REV = self.rev or self.dirtyRev or null;
-          });
-        helix = makeOverridableHelix self.packages.${system}.helix-unwrapped {};
-        default = self.packages.${system}.helix;
-      };
+        strictDeps = true;
 
-      checks = {
-        # Build the crate itself
-        inherit (self.packages.${system}) helix;
+        # Sets the Helix runtimedir to the grammars
+        env.HELIX_DEFAULT_RUNTIME = "${runtimeDir}";
 
-        clippy = craneLibMSRV.cargoClippy (commonArgs
-          // {
-            inherit cargoArtifacts;
-            cargoClippyExtraArgs = "--all-targets -- --deny warnings";
-          });
+        # Get all the application stuff in the output directory.
+        postInstall = ''
+          mkdir -p $out/lib
+          installShellCompletion ${./contrib/completion}/hx.{bash,fish,zsh}
+          mkdir -p $out/share/{applications,icons/hicolor/256x256/apps}
+          cp ${./contrib/Helix.desktop} $out/share/applications
+          cp ${./contrib/helix.png} $out/share/icons/hicolor/256x256/apps
+        '';
 
-        fmt = craneLibMSRV.cargoFmt commonArgs;
+        meta.mainProgram = "hx";
+      });
+  in
+    flake-utils.lib.eachDefaultSystem (system: let
+      pkgs = import nixpkgs {
+        inherit system;
+        overlays = [(import rust-overlay)];
+      };
 
-        doc = craneLibMSRV.cargoDoc (commonArgs
-          // {
-            inherit cargoArtifacts;
-          });
+      # Get Helix's MSRV toolchain to build with by default.
```

Comment:
For the build of `packages.helix` we can use the latest stable - that's what we do in the release builds as well. We only need the MSRV toolchain for the development shell